shared/pub/metadata.js

```diff
@@ -0,0 +1,54 @@
+export const getPDFDownloads = (pub) => {
+	const downloads = pub.downloads;
+	const exports = pub.activeBranch.exports;
+	if (downloads && downloads.length > 0) {
+		return downloads.filter((download) => {
+			if (download.url.substring(download.url.length - 4) === '.pdf') {
+				return download;
+			}
+			return false;
+		});
+	}
+	if (exports && exports.length > 0) {
+		return exports.filter((exportFile) => {
+			if (exportFile.format === 'pdf') {
+				return exportFile;
+			}
+			return false;
+		});
+	}
+	return false;
+};
+
+export const getTextAbstract = (content) => {
+	let abstract = '';
+	if (!content) {
+		return abstract;
+	}
+	if (
```

Comment:
⚠️ This is missing an existence check for `content.content[...]`. The document may have only one (or maybe even zero?) entries in the `content` array.